aten/src/ATen/TensorUtils.cpp

```diff
@@ -132,6 +132,31 @@ void checkAllSameGPU(CheckedFrom c, ArrayRef<TensorArg> tensors) {
   checkAllSame(c, tensors, checkSameGPU);
 }
 
+void checkSameHB(CheckedFrom c, const TensorArg& t1, const TensorArg& t2) {
+  if (! (t1->is_hammerblade()) || ! (t2->is_hammerblade())) {
+    std::ostringstream oss;
+    if (! t1->is_hammerblade()) {
+      oss << "Tensor for " << t1 << " is not on HB, ";
+    }
+    if (! t2->is_hammerblade()) {
+      oss << "Tensor for " << t2 << " is not on HB, ";
+    }
+    oss << "but expected " 
+        << ((!(t1->is_hammerblade() || t2->is_hammerblade())) ? "them" : "it")
+        << " to be on HB (while checking arguments for " << c << ")";
+    AT_ERROR(oss.str());
+  }
+  TORCH_CHECK(
+    t1->get_device() == t2->get_device(),
```

Comment:
Currently HammerBlade is hardcoded to have device index == -1. But I think there is no harm to check it, for extensibility.